crates/bevy_app/src/plugin.rs

```diff
@@ -90,6 +150,35 @@ pub trait Plugin: Downcast + Any + Send + Sync {
     fn is_unique(&self) -> bool {
         true
     }
+
+    /// Updates the plugin to a desired [`PluginState`].
+    fn update(&mut self, app: &mut App, state: PluginState) {
+        match state {
+            PluginState::Init => self.init(app),
+            PluginState::Building => self.build(app),
+            PluginState::Configuring => self.configure(app),
+            PluginState::Finalizing => self.finalize(app),
+            PluginState::Done => {}
+            s => panic!("Cannot handle {s:?} state"),
```

Comment:
We can just remove this arm, correct?

Reply:
Idle and cleanup are the two states that cannot be processed. Is it better to add both to the match?